generate-geodatabase-replica-from-feature-service/src/main/java/com/esri/arcgismaps/sample/generategeodatabasereplicafromfeatureservice/MainActivity.kt

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright 2023 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.generategeodatabasereplicafromfeatureservice
+
+import android.app.AlertDialog
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.lifecycleScope
+import com.arcgismaps.ApiKey
+import com.arcgismaps.ArcGISEnvironment
+import com.arcgismaps.Color
+import com.arcgismaps.data.Geodatabase
+import com.arcgismaps.geometry.Envelope
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.layers.FeatureLayer
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.tasks.geodatabase.GenerateGeodatabaseJob
+import com.arcgismaps.tasks.geodatabase.GeodatabaseSyncTask
+import com.esri.arcgismaps.sample.generategeodatabasereplicafromfeatureservice.databinding.ActivityMainBinding
+import com.esri.arcgismaps.sample.generategeodatabasereplicafromfeatureservice.databinding.GenerateGeodatabaseDialogLayoutBinding
+import com.google.android.material.snackbar.Snackbar
+import kotlinx.coroutines.launch
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG = MainActivity::class.java.simpleName
+
+    // set up data binding for the activity
+    private val activityMainBinding: ActivityMainBinding by lazy {
+        DataBindingUtil.setContentView(this, R.layout.activity_main)
+    }
+
+    // setup data binding for the mapview
+    private val mapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    // starts the geodatabase replica process
+    private val generateButton by lazy {
+        activityMainBinding.generateButton
+    }
+
+    // shows the geodatabase loading progress
+    private val progressDialog by lazy {
+        GenerateGeodatabaseDialogLayoutBinding.inflate(layoutInflater)
+    }
+
+    // local file path to the geodatabase
+    private val geodatabaseFilePath by lazy {
+        getExternalFilesDir(null)?.path + getString(R.string.portland_trees_geodatabase_file)
```

Comment:
This is something I saw recently across samples, it's a nitpick though. I think generally it's probably better to use `File(getExternalFilesDir(null)?, getString(R.string.portland_trees_geodatabase_file)` because it would automatically handle the file separators and you can verify that the file exists etc. However in this case it's probably better to match the general pattern of the samples so there's not one sample using `File` and the rest aren't.

That said, does this line need a separator?
```suggestion
        getExternalFilesDir(null)?.path + File.separator() +  getString(R.string.portland_trees_geodatabase_file)
```

Reply:
I think @hud10837 is reffering to the call to set the `provisionPath` once the library downloads the portal item where the use of `File.separator` is required:
```kotlin
private val provisionPath: String by lazy {
    getExternalFilesDir(null)?.path.toString() + File.separator + getString(R.string.app_name)
}
```
Since this is not related to the library, in general, it's okay to add the line separator to the string filename instead 👍